### PR TITLE
make some function params optional

### PR DESCRIPTION
**Pull request by some-boi**
if all the properties in it is optional then why is it required

### CODE REVIEW

**Comment by itohatweb** [comment]
What's the reasoning of making them optional?

**Comment by some-boi** [comment]
> What's the reasoning of making them optional?

~~annoying~~
ya ive been checking the code  and realized it so i decided to fix that stuff even tho it does nothing so idk

**Comment by itohatweb** [comment]
> > What's the reasoning of making them optional?
> 
> ~annoying~ ya ive been checking the code and realized it so i decided to fix that stuff even tho it does nothing so idk

I understand now. 
Please make this a breaking change hehe. Move the guild icon hash stuff out of the options and make it an undefinable param (`icon?: string | bigint`). 
For avatar urls there can be discussed whats the better method.
Please revert the changes from `editBotProfile` and open an issue indicating that the user should provide at least one option of it. (PS: if you want to figure the types out you can do this of course.)

**Review by lts20050703 — CHANGES_REQUESTED** [review]
<https://deno.land/manual@v1.16.4/contributing/style_guide#exported-functions-max-2-args-put-the-rest-into-an-options-object>
avatarUrl.ts ✅
guildIconUrl.ts and guildSplashUrl.ts: As much as it annoys you, it annoys more people having to deal with this breaking change and 4 parameters is too much.

**Comment by itohatweb** [comment]
4 options are ok for us. It should not be more than 4 though. V13 is already a big change this won't change much. Also not many ppl use it I would say.
imho

**Comment by lts20050703** [comment]
@some-boi If you have a discord account consider joining https://discord.gg/RxaygRBt
image

**Comment by itohatweb** [comment]
What is this Discord Server invite link 🤔
It's not discordeno

**Comment by Yaikava** [comment]
> What is this Discord Server invite link 🤔 It's not discordeno

It is discordeno.

**Comment by itohatweb** [comment]
Lol for me it looked like a vanity invite 🤣

**Comment by some-boi** [comment]
> 



> @some-boi If you have a discord account consider joining https://discord.gg/RxaygRBt image

but im there (a boi here ;o#6969 or by nick Major General Henry Landry)

**Comment by lts20050703** [comment]
> > 
> 
> > @some-boi If you have a discord account consider joining https://discord.gg/RxaygRBt image
> 
> but im there (a boi here ;o#6969 or by nick Major General Henry Landry)

oh... oops sorry I dont know your discord username and you're already in there lol

**Review by Skillz4Killz — CHANGES_REQUESTED** [review]

File: src/helpers/guilds/guildIconUrl.ts [hunk 1]
```diff
@@ -6,20 +6,20 @@ import type { Bot } from "../../bot.ts";
 export function guildIconURL(
   bot: Bot,
   id: bigint,
-  options: {
-    icon?: string | bigint;
+  icon: string | bigint | undefined,
```
[the remark this answers is not in the document]
Comment (Skillz4Killz) [reply src/helpers/guilds/guildIconUrl.ts:9]: @itohatweb

[the remark this answers is not in the document]
Comment (lts20050703) [reply src/helpers/guilds/guildIconUrl.ts:9]: imo if we're gonna make v13 ~~"the nuke update"~~ "the functional and bigint update" then imo we should break backward compatibility

[the remark this answers is not in the document]
Comment (some-boi) [reply src/helpers/guilds/guildIconUrl.ts]: i also dont see a point on it since the types are know bigint | undefined

**Review by Skillz4Killz — APPROVED** [review]

**Comment by Skillz4Killz** [comment]
Great PR!

**Pull request merged**